source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -11,150 +12,203 @@
             }
+        }
 
-            public Task<TResource> Get(string idOrHref)
-            {
-                if (string.IsNullOrWhiteSpace(idOrHref))
-                    return null;
+        bool IsInSingleSpaceContext()
+        {
+            var spaceIds = GetSpaceIds();
+            var isASpecificSpaceId = spaceIds.Length == 1 && !spaceIds.Contains("all");
+            return isASpecificSpaceId && !IsIncludingSystem();
+        }
 
-                var parameters = ParameterHelper.CombineParameters(AdditionalQueryParameters, new { id = idOrHref });
-                return idOrHref.StartsWith("/", StringComparison.OrdinalIgnoreCase)
-                    ? Client.Get<TResource>(idOrHref, AdditionalQueryParameters)
-                    : Client.Get<TResource>(Client.Link(CollectionLinkName), parameters);
-            }
+        string[] GetSpaceIds()
+        {
+            var isMixedScope = TypeUtil.IsAssignableToGenericType(this.GetType(), typeof(ICanExtendSpaceContext<>));
+            return isMixedScope ? AdditionalQueryParameters["spaces"] as string[] : Client.SpaceContext.SpaceIds.ToArray();
+        }
 
-            public virtual async Task<List<TResource>> Get(params string[] ids)
-            {
-                if (ids == null) return new List<TResource>();
-                var actualIds = ids.Where(id => !string.IsNullOrWhiteSpace(id)).ToArray();
-                if (actualIds.Length == 0) return new List<TResource>();
-
-                var resources = new List<TResource>();
-                var link = Client.Link(CollectionLinkName);
-                if (!Regex.IsMatch(link, @"\{\?.*\Wids\W"))
-                    link += "{?ids}";
-
-                var parameters = ParameterHelper.CombineParameters(AdditionalQueryParameters, new { ids = actualIds });
-                await Client.Paginate<TResource>(
-                    link,
-                    parameters,
-                    page =>
-                    {
-                        resources.AddRange(page.Items);
-                        return true;
-                    })
-                    .ConfigureAwait(false);
-
-                return resources;
-            }
+        bool IsIncludingSystem()
+        {
+            var isMixedScope = TypeUtil.IsAssignableToGenericType(this.GetType(), typeof(ICanExtendSpaceContext<>));
+            return isMixedScope
+                ? bool.Parse(AdditionalQueryParameters["includeSystem"].ToString())
+                : Client.SpaceContext.IncludeSystem;
+        }
 
-            public Task<TResource> Refresh(TResource resource)
+        void ValidateSpaceId(TResource resource)
+        {
+            if (resource is IHaveSpaceResource spaceResource)
             {
-                if (resource == null) throw new ArgumentNullException("resource");
-                return Get(resource.Id);
+                var spaceIds = GetSpaceIds();
+                var isWildcard = spaceIds != null && spaceIds.Contains("all");
```

Comment:
another case of `"all"`

can you name this differently, we have a concept of wildcard in permissions, just so there's no way to confuse the concepts.